src/SprykerEco/Zed/CrefoPay/Business/Oms/Command/Builder/CaptureOmsCommandRequestBuilder.php

```diff
@@ -97,12 +94,32 @@ protected function isFirstCapture(CrefoPayOmsCommandTransfer $crefoPayOmsCommand
      */
     protected function addExpensesRequest(CrefoPayOmsCommandTransfer $crefoPayOmsCommandTransfer): CrefoPayOmsCommandTransfer
     {
+        if ($this->config->getCaptureExpensesSeparately() && !$this->isFirstCapture($crefoPayOmsCommandTransfer)) {
+            return $crefoPayOmsCommandTransfer;
+        }
+
         $expensesAmountToCapture = $this->calculateExpensesAmount($crefoPayOmsCommandTransfer->getOrder());
 
         if ($expensesAmountToCapture <= 0) {
             return $crefoPayOmsCommandTransfer;
         }
 
+        if (!$this->config->getCaptureExpensesSeparately()) {
+            $orderItemsAmountToCapture = $crefoPayOmsCommandTransfer
```

Comment:
This methods became quite big. Could you put this to the separate method?

Reply:
Refactored.